runtime-modules/content/src/lib.rs

```diff
@@ -553,6 +592,31 @@ decl_module! {
             Self::deposit_event(RawEvent::ChannelUpdated(actor, channel_id, channel, params));
         }
 
+        // Extrinsic for updating channel privilege level (requires lead access)
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn update_channel_privilege_level(
+            origin,
+            channel_id: T::ChannelId,
+            new_privilege_level: T::ChannelPrivilegeLevel,
+        ) {
+            let sender = ensure_signed(origin)?;
+
+            ensure_lead_auth_success::<T>(&sender)?;
+
+            // check that channel exists
+            let mut channel = Self::ensure_channel_validity(&channel_id)?;
```

Comment:
In general, I try to encourage everyone to avoid having mutables with any sort of longer life-time, as it just opens to the door to some bad mutation being introduced at some later time. This case is really mild, so not a big issue, but try to avoid when possible, like using ::mutate storage action, or constructing a new entity object using the old one combined with struct update syntax
https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax

So here it would be

```Rust
let new_channel = Channel {
privilege_level: new_privilege_level
...old_channel
}
```

Reply:
Done in https://github.com/Joystream/joystream/pull/3366/commits/4ef05ccc660879617d616332926e3cb3cd03082f